File: ios/Freedom/Core/Data/MessageData.swift
// ═════════════════════════════════════════════════════════════════════════════
//  MessageData — Chat message entity mirroring Android Room schema
// ═════════════════════════════════════════════════════════════════════════════

import Foundation
import GRDB

struct MessageData: Codable, FetchableRecord, MutablePersistableRecord, Identifiable, Equatable {
    static let databaseTableName = "message_data"

    static let SENT     = "SENT"
    static let RECEIVED = "RECEIVED"

    var id: Int64?
    var timestamp: String?
    var messageType: String?
    var content: String?
    var sender: String?
    var contactId: Int64 = 0
    var direction: String = MessageData.RECEIVED

    enum CodingKeys: String, CodingKey {
        case id
        case timestamp
        case messageType = "message_type"
        case content
        case sender
        case contactId = "contact_id"
        case direction
    }

    mutating func didInsert(_ inserted: InsertionSuccess) {
        id = inserted.rowID
    }
}

File: main/src/ui/java/freedom/app/data/entity/MessageData.kt
package freedom.app.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "message_data")
data class MessageData(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    var id: Long = 0,

    @ColumnInfo(name = "timestamp")
    var timestamp: String?,

    @ColumnInfo(name = "message_type")
    var messageType: String?,

    @ColumnInfo(name = "content")
    var content: String?,

    @ColumnInfo(name = "sender")
    var sender: String?,

    /** DB id of the contact this message belongs to — 0 = unknown/legacy. */
    @ColumnInfo(name = "contact_id")
    var contactId: Long = 0,

    /** SENT = I sent this; RECEIVED = I received this. */
    @ColumnInfo(name = "direction")
    var direction: String = RECEIVED
) {
    companion object {
        const val SENT     = "SENT"
        const val RECEIVED = "RECEIVED"
    }
}
